packages/express/src/clerkMiddleware.ts

```diff
@@ -0,0 +1,43 @@
+import { createClerkClient } from '@clerk/backend';
+import type { RequestHandler } from 'express';
+
+import { authenticateRequest, setResponseHeaders } from './authenticateRequest';
+import { clerkClient as defaultClerkClient } from './clerkClient';
+import type { ClerkMiddleware, ClerkMiddlewareOptions } from './types';
+import { defaultHandler } from './utils';
+
+const parseHandlerAndOptions = (args: unknown[]) => {
+  return [
+    typeof args[0] === 'function' ? args[0] : undefined,
+    (args.length === 2 ? args[1] : typeof args[0] === 'function' ? {} : args[0]) || {},
+  ] as [RequestHandler | undefined, ClerkMiddlewareOptions];
+};
+
+export const clerkMiddleware: ClerkMiddleware = (...args: unknown[]): RequestHandler[] => {
```

Comment:
JSDoc description here would be great

Reply:
I have added JSDoc in the `ClerkMiddleware` definition, just like we did in our `@clerk/nextjs` package. Do you think we should move those to the actual implementation?
Our nextjs JSDocs, used as example: https://github.com/clerk/javascript/blob/4565ef29e67d4e39f14c7e61c71fb411fab9aa8e/packages/nextjs/src/server/clerkMiddleware.ts#L41-L63

Reply:
ty for the feedback. I will move it and add a ticket to move this for our nextjs package.
cc: @nikosdouvlis

Reply:
@LekoArts Checking again your screenshot above,  we can see that there is an `@example` so i guess that the JSDoc work.

Reply:
I'd consider that example useless as it just shows you what you're already using at the moment, not the other options or more details.
It also doesn't have any description or the params explained.

If you put the description and all three examples onto the function, not the type the IntelliSense will be better

Reply:
I have moved the JSDocs above code definition.
<img width="992" alt="Screenshot 2024-03-15 at 15 53 28" src="https://github.com/clerk/javascript/assets/3936408/e40eb7b7-cc5f-4e16-89ac-0ef12bda4268">